### DISCUSSION TITLE
mounting disk fails

### DISCUSSION THREAD

**Discussion by till** [flatcar/Flatcar#1201] [Q&A] [answered]
I just ran into an odd problem with mounting a disk, I followed [the example on the docs how to (bind) mount storage](https://www.flatcar.org/docs/latest/setup/storage/mounting-storage/#use-attached-storage-for-docker) and it seems the unit is off. Should I turn this into a ticket?

Anyway, here are the steps:

1. I am creating a disk and I am creating an instance (on OpenStack)
2. I configure ignition/butane to label the disk
3. I am adding a unit (also via ignition/butane), but that produces an error

Here is some output:

```
node-001 /home/core # ls -lah /dev/disk/by-label/
total 0
drwxr-xr-x.  2 root root 140 Oct  6 13:06 .
drwxr-xr-x. 10 root root 200 Oct  6 12:15 ..
lrwxrwxrwx.  1 root root   9 Oct  6 12:15 CUSTOMER -> ../../sdb
lrwxrwxrwx.  1 root root  10 Oct  6 13:06 EFI-SYSTEM -> ../../sda1
lrwxrwxrwx.  1 root root  10 Oct  6 13:06 OEM -> ../../sda6
lrwxrwxrwx.  1 root root  10 Oct  6 13:06 ROOT -> ../../sda9
lrwxrwxrwx.  1 root root   9 Oct  6 12:15 config-2 -> ../../sr0
```

The disk is labeled `CUSTOMER` ✅ 

But the mount unit doesn't work:

```
node-001 /home/core # systemctl cat k0s-storage.mount
# /etc/systemd/system/k0s-storage.mount
[Unit]
Description=Mount CUSTOMER to /var/lib/k0s
Before=local-fs.target
[Mount]
What=/dev/disk/by-label/CUSTOMER
Where=/var/lib/k0s
Type=ext4
[Install]
WantedBy=local-fs.target 
```

```
core@node-001 ~ $ sudo systemctl status k0s-storage.mount
○ k0s-storage.mount - Mount CUSTOMER to /var/lib/k0s
     Loaded: bad-setting (Reason: Unit k0s-storage.mount has a bad unit file setting.)
     Active: inactive (dead)
      Where: /var/lib/k0s
       What: /dev/disk/by-label/CUSTOMER

Oct 06 13:14:05 node-001.db-prod systemd[1]: k0s-storage.mount: Cannot add dependency job, ignoring: Exec format error
Oct 06 13:14:05 node-001.db-prod systemd[1]: k0s-storage.mount: Cannot add dependency job, ignoring: Unit k0s-storage.mount has a bad unit file setting.
Oct 06 13:14:06 node-001.db-prod systemd[1]: k0s-storage.mount: Cannot add dependency job, ignoring: Unit k0s-storage.mount has a bad unit file setting.
Oct 06 13:14:06 node-001.db-prod systemd[1]: k0s-storage.mount: Cannot add dependency job, ignoring: Unit k0s-storage.mount has a bad unit file setting.
Oct 06 13:14:07 node-001.db-prod systemd[1]: k0s-storage.mount: Cannot add dependency job, ignoring: Unit k0s-storage.mount has a bad unit file setting.
Oct 06 13:14:07 node-001.db-prod systemd[1]: k0s-storage.mount: Cannot add dependency job, ignoring: Unit k0s-storage.mount has a bad unit file setting.
Oct 06 13:15:46 node-001.db-prod systemd[1]: k0s-storage.mount: Cannot add dependency job, ignoring: Unit k0s-storage.mount has a bad unit file setting.
Oct 06 13:15:46 node-001.db-prod systemd[1]: k0s-storage.mount: Cannot add dependency job, ignoring: Unit k0s-storage.mount has a bad unit file setting.
Oct 06 13:15:46 node-001.db-prod systemd[1]: k0s-storage.mount: Cannot add dependency job, ignoring: Unit k0s-storage.mount has a bad unit file setting.
Oct 06 13:21:29 node-001.db-prod systemd[1]: k0s-storage.mount: Where= setting doesn't match unit name. Refusing.
```

My guess was that `/var/lib/k0s` must exist, which I maybe thought would be handled by the unit. But even when I created it, it doesn't seem to work. Is there anything obvious I am missing?

**Comment by jepio** [comment, marked as answer]
Hi @till 

from https://www.freedesktop.org/software/systemd/man/systemd.mount.html:
> Mount units must be named after the mount point directories they control. Example: the mount point /home/lennart must be configured in a unit file home-lennart.mount. For details about the escaping logic used to convert a file system path to a unit name, see [systemd.unit(5)](https://www.freedesktop.org/software/systemd/man/systemd.unit.html#). Note that mount units cannot be templated, nor is possible to add multiple names to a mount unit by creating symlinks to its unit file.

So you need to name your unit `var-lib-k0s.mount`

**Reply by till** [reply]
Argh... I knew it must have been simple. Thank you!
